Modules/Core/Sources/Core/Buffer/Buffer.swift

```diff
@@ -6,7 +6,136 @@ import Darwin
 #endif
 
 public typealias Byte = UInt8
-public typealias Buffer = DispatchData
+
+public struct Buffer : RandomAccessCollection {
+    public typealias Iterator = BufferIterator
+    public typealias Index = Int
+    public typealias Indices = DefaultRandomAccessIndices<Buffer>
+    
+    public private(set) var bytes: ContiguousArray<Byte>
+    
+    public var count: Int {
+        return bytes.count
+    }
+    
+    public init(_ bytes: [Byte] = []) {
+        self.bytes = ContiguousArray(bytes)
+    }
+    
+    public init(bytes: [Byte]) {
+        self.bytes = ContiguousArray(bytes)
+    }
+    
+    public init(bytes buffer: UnsafeBufferPointer<UInt8>) {
+        self.bytes = ContiguousArray(buffer)
+    }
+    
+    public mutating func append(_ other: Buffer) {
+        bytes += other.bytes
+    }
+    
+    public mutating func append(_ other: [Byte]) {
+        bytes += other
+    }
+    
+    public mutating func append(_ other: UnsafeBufferPointer<Byte>) {
+        guard other.count > 0 else {
+            return
+        }
+        bytes += [Byte](other)
+    }
+    
+    public mutating func append(_ other: UnsafePointer<Byte>, count: Int) {
+        guard count > 0 else {
+            return
+        }
+        bytes += [UInt8](UnsafeBufferPointer(start: other, count: count))
+    }
+    
+    public subscript(index: Index) -> Byte {
+        return bytes[index]
+    }
+    
+    public subscript(bounds: Range<Int>) -> Buffer {
+        return Buffer(bytes: [Byte](bytes[bounds]))
```

Comment:
why not return a slice here? sometimes you don't need a copy. maybe its better to leave the decision to the calling site.